### DISCUSSION TITLE
Same optimization runs return marginally different results

### DISCUSSION THREAD

**Discussion by anubhavdas0907** [micom-dev/micom#68] [Q&A] [answered]
Hello,

Thanks for developing this nice package.

My query is regarding the randomness in the results. Optimization runs on same models, return different results.
Although the differences are marginal, in some cases, the growth rates of some taxa are reversed in the second run.

I understand, there might be a stochastic element in the analysis, but kindly suggest a way to make the output consistent for reproducibility.

Regards

**Comment by cdiener** [comment, marked as answer]
Hm, yeah that should not happen. There is not really a stochastic element to the analysis but the numerical solver do have an inherent accuracy limit. For Cplex this is around 1e-5 to 1e-6 in practice and for OSQP around 1e-3 to 1e-4. If the discrepancies are in that range it is a numerical accuracy issue. However, it may not be. So it would be great if you could give some more information what code exactly you are running and with which solver.

**Reply by anubhavdas0907** [reply]
Hello Christian,
Thanks for your reply. I will keep a note of that.
Regarding my issue, I realized I was comparing solutions optimized using different solvers.

I have another question, regarding increasing the speed of the computation. Is it possible to parallelize the optimization command. I will be using OSQP as a solver.

Thanks and Regards

**Reply by cdiener** [reply]
For a single model, no. But if you are processing different samples then yes. Look at the `grow` workflow for instance.
